t/ui/24-version.t

```diff
@@ -0,0 +1,53 @@
+# Copyright (C) 2017 SUSE Linux Products GmbH
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+BEGIN {
+    unshift @INC, 'lib';
+}
+
+use Mojo::Base -strict;
+use Test::More;
+use Test::Mojo;
+use Test::Warnings;
+use OpenQA::Test::Case;
+
+OpenQA::Test::Case->new->init_data;
+my $t = Test::Mojo->new('OpenQA::WebAPI');
+
+my $changelog_file = $t->app->home->child('public/Changelog');
+my $refs_file      = $t->app->home->child('.git/packed-refs');
+my $check_regex    = qr/Version/;
+my $version_regex  = qr/Version(\h*\R*)+/;
+my $sha_regex      = qr/\b[0-9a-f]{5,40}\b/;
+my $t_func         = "text_like";
+
+# Let's check that the version matches our versioning scheme.
+# If it's a git version it should be in the form: git-tag-sha1
+# otherwise is a group of 3 decimals followed by a partial sha1: a.b.c.sha1
+if ($changelog_file and -e $changelog_file) {
```

Comment:
we could also use an empty inline package with just an "app" sub that returns a Mojo::Home object, but would be glad to see if there is a more straightforward solution